crates/biome_js_analyze/src/lint/nursery/no_restricted_imports.rs

```diff
@@ -54,16 +69,40 @@ impl Rule for NoRestrictedImports {
 
     fn run(ctx: &RuleContext<Self>) -> Self::Signals {
         let node = ctx.query();
+        let file_path = ctx.file_path();
+
         if node.is_in_ts_module_declaration() {
             return None;
         }
+
+        // "import { merge } from 'lodash';" => get the "lodash"
         let module_name = node.module_name_token()?;
         let inner_text = inner_string_text(&module_name);
 
         ctx.options()
             .paths
-            .get(inner_text.text())
-            .map(|message| (module_name.text_trimmed_range(), message.to_string()))
+            .iter()
+            .filter_map(|(path, config)| {
+                if !inner_text.text().starts_with(path) {
+                    return None;
+                }
+
+                if let Some(allowed_from) = &config.allowed_from {
+                    if allowed_from.contains(&file_path.to_string_lossy().to_string()) {
```

Comment:
A couple of things on this line:
- `.contains` can create false positives. For example, if your configuration is `./src/something.js`, a path like `../src/something.js` will pass, which is not what you want, right?
- I don't think we need to allocate a `String` (via `.to_string`) to do a simple check

Reply:
About the the allocation of the ```.to_string```, I can use:

```
file_path.to_string_lossy().matches(allowed).count() > 0
```


And about the false positives, it would be nice to allow something like "./src/**/tests" or other patterns. To allow this, I would need to know the working directory root. because ```file_path``` returns the complete path, and to use the "./" notation as you pointed, I'd need to convert it. Is there any helpers to do that?